src/events/tradeEvents.js

```diff
@@ -15,32 +15,26 @@ const { getTrade, getOldTrade } = require('../services/getTrade');
  */
 module.exports.findTradeEvents = async (timestamp, pastTimeInSeconds) => {
     console.log(`Looking for new trades at ${timestamp}`);
-    getTrade(timestamp, pastTimeInSeconds, 20)
-        .then(trades => {
-            trades.forEach(trade => {
-                const message = new Array();
-                const type = (trade.type === 'Buy') ? 'purchased' : 'sold';
-                const odds = parseFloat(trade.outcomeTokenMarginalPrices[trade.outcomeIndex] * 100 ).toFixed(2);
-                Promise.all([
-                    getTokenName(web3, trade.collateralToken),
-                    getTokenDecimals(web3, trade.collateralToken),
-                    getOldTrade(trade.id),
-                ])
-                    .then(([tokenName, decimals, oldTrade]) => {
-                        const amount = parseFloat(trade.collateralAmount / 10**decimals).toFixed(2);
-                        if (trade.collateralAmountUSD > 1000.00) {
-                            message.push('<!here>');
-                        }
-                        const oldOdds = oldTrade ? parseFloat(oldTrade.outcomeTokenMarginalPrices[trade.outcomeIndex] * 100 ).toFixed(2) : '0.00';
-                        const outcome = trade.outcomes ? trade.outcomes[trade.outcomeIndex] : trade.outcomeIndex;
-                        message.push(`> ${amount} <https://${urlExplorer}/token/${trade.collateralToken}|${tokenName}> of *${outcome}* ${type} in "<https://omen.eth.link/#/${trade.fpmm}|${trade.title}>".`,
-                            `> Outcome odds: ${oldOdds}% --> ${odds}%`,
-                            `> *Created by*: <https://omen.eth.link/#/${trade.creator}|${truncate(trade.creator, 14)}>`,
-                        );
-                        // Send Slack notification
-                        pushSlackArrayMessages(message);
-                        console.log(message.join('\n') + '\n');
-                    });
-            });
-        });
+    const trades = await getTrade(timestamp, pastTimeInSeconds, 20)
+    trades.forEach(async trade => {
```

Comment:
If you want to execute some tasks in sequence, you cannot use forEach indeed. Just use a modern [for … of loop](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Sentencias/for...of) instead, in which await will work as expected

An example
```
async function foo () {
  const elements = await doSomething()

  for (const element of elements) {
    const bar = await somethingAgain()
    console.log(bar)
  }
}
```